### PR TITLE
Improve LIT tests

### LINKED ISSUE #169: Improve test coverage
**Description:**

Revisit existing LIT tests and improve coverage; including negative test cases where applicable 

**Describe your problem(s)**

**Describe your solution(s)**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

Ruvi-d

### LINKED ISSUE #199: Boolean values appear to be always evaluated as true
**Description:**

The following test fails
```
// RUN: "%testRunScript" %s %nballerinacc "%java_path" | filecheck %s

public function print_str(string val) = external;

public function printu32(int val) = external;

public function main() {
    boolean f = false;
    print_str("RESULT=");
    if (f) {
        printu32(0);
    }
    else {
        printu32(42);
    }
}
// CHECK: RESULT=42
```

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

Priority/Blocker

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by ruvi-d**
## Purpose

- Add new test cases & reorganize existing tests
- Renamed folders and test file names according to the guidelines at : https://github.com/ballerina-platform/ballerina-distribution/blob/master/doc-guidelines.md#ballerina-by-examples-guidelines
- Add support to optionally skip BIR generation when running tests (refer updated readme for instructions)
- Add support to filter tests to run via regex (refer updated readme for instructions)
- Fix bug with boolean constant loading
- Removes special casing for _bal_result global variable

Resolves #169 
Resolves #199 

## Approach
Top level folders were created for each type. The test file names were rename following the `<action>-<type>-<other>.bal` format. This allows developers to easily identify missing tests. 
New tasks will be created to track disabled and missing tests due to incomplete features or bugs.

## Related PRs
Depends on #187

### CODE REVIEW

File: test/int/function_call_global_int.bal [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// RUN: "%testRunScript" %s %nballerinacc "%java_path" "%skip_bir_gen" | filecheck %s
+
+public function print_string(string val) = external;
+
+public function printu32(int val) = external;
+
+public function bar(int z) returns int
+{
+    return z + 10;
+}
```
Comment (KavinduZoysa) [inline test/int/function_call_global_int.bal:10]: Format error, This`{` should be inline with the function definition.

Comment (KavinduZoysa) [reply test/int/function_call_global_int.bal:10]: Please check other places too.

Comment (ruvi-d) [reply test/int/function_call_global_int.bal:10]: noted thanks.

Comment (ruvi-d) [reply test/int/function_call_global_int.bal:10]: Looks like `ballerina format` doesn't fix that.
Anyway, ran ballerina format and committed.

File: compiler/codegen/TypeCastInsn.cpp [hunk 2]
```diff
@@ -39,12 +39,10 @@ void TypeCastInsn::translate(LLVMModuleRef &modRef) {
...
-    auto lhsVar = funcObj.getLocalOrGlobalVariable(getLhsOperand());
-    assert(lhsVar.has_value());
-    auto rhsVar = funcObj.getLocalOrGlobalVariable(rhsOp);
-    assert(rhsVar.has_value());
-    auto lhsType = lhsVar->getType().getTypeTag();
-    auto rhsType = rhsVar->getType().getTypeTag();
+    const auto &lhsVar = funcObj.getLocalOrGlobalVariable(getLhsOperand());
+    const auto &rhsVar = funcObj.getLocalOrGlobalVariable(rhsOp);
+    auto lhsType = lhsVar.getType().getTypeTag();
+    auto rhsType = rhsVar.getType().getTypeTag();
```
Comment (KavinduZoysa) [inline compiler/codegen/TypeCastInsn.cpp:45]: Shall we rename these variables as `lhsTypeTag` and `rhsTypeTag`?

**Comment by KavinduZoysa** [comment]
@ruvi-d, I have seen a couple of test cases are skipped and a couple of TODOs. Do we have git issues to track them?

File: compiler/codegen/TypeCastInsn.cpp (hunk 2 above)
Comment (ruvi-d) [reply compiler/codegen/TypeCastInsn.cpp:45]: That rename change is there in #200

**Comment by ruvi-d** [comment]
> @ruvi-d, I have seen a couple of test cases are skipped and a couple of TODOs. Do we have git issues to track them?

We have issues created for the reasons behind all the todo's and skipped tests. Which include #215 , #214 and a few others.
Is that what you are looking for?

**Comment by KavinduZoysa** [comment]
> > @ruvi-d, I have seen a couple of test cases are skipped and a couple of TODOs. Do we have git issues to track them?
> 
> We have issues created for the reasons behind all the todo's and skipped tests. Which include #215 , #214 and a few others.
> Is that what you are looking for?

ACK

**Review by KavinduZoysa — APPROVED** [review]

**Review by ramakotareddy — APPROVED** [review]

**Pull request merged**